### PR TITLE
Updates to support sharding

### PR DESCRIPTION
**Pull request by abdulrabbani00**
@ashwinphatak @nikugogoi @prathamesh0 - Just wanted to bring this to your individual attention prior to merging the change to develop.

This change will require updates to your `config.sh` file if you are using either `geth` or the `ipld-eth-db`. Please let me know if you have any questions. I'll give you guys a few days to test this if you'd like before merging.

### CODE REVIEW

**Review by prathamesh0 — COMMENTED** [review]
We can avoid use of variables by just having v3 and v4 db setup in separate docker-compose files in the same branch. We can keep the same db name and db host name in both v3 and v4 setups for minimal difference in v3 and v4 docker-compose files for other services using the db.
See https://github.com/vulcanize/ipld-eth-db/pull/88 and https://github.com/vulcanize/stack-orchestrator/pull/25.

**Comment by abdulrabbani00** [comment]
@prathamesh0 - If this looks good to you, please feel free to squash and merge. I just wanted you to take a final look over it before merging to `main`.

File: docker/latest/docker-compose-ipld-eth-server.yml [hunk 1]
```diff
@@ -2,22 +2,26 @@ version: "3.2"
 services:
   ipld-eth-server:
     restart: unless-stopped
+    depends_on:
+      - ipld-eth-db
     image: vulcanize/ipld-eth-server:v2.0.0
```
Comment (nikugogoi) [inline docker/latest/docker-compose-ipld-eth-server.yml]: Can update version to [v3.2.2](https://hub.docker.com/layers/ipld-eth-server/vulcanize/ipld-eth-server/v3.2.2/images/sha256-bacb17c6654b0f3587e7eef42c052aa0311304e8919cd57e2f23458fb7c709d3?context=explore)

File: docker/latest/docker-compose-ipld-eth-server.yml [hunk 2]
```diff
@@ -2,22 +2,26 @@ version: "3.2"
...
+      DATABASE_HOSTNAME: ipld-eth-db
       DATABASE_PORT: 5432
       DATABASE_USER: "vdbm"
       DATABASE_PASSWORD: "password"
       ETH_CHAIN_ID: 99
       ETH_FORWARD_ETH_CALLS: $eth_forward_eth_calls
       ETH_PROXY_ON_ERROR: $eth_proxy_on_error
       ETH_HTTP_PATH: $eth_http_path
+      WATCHED_ADDRESS_GAP_FILLER_ENABLED: $watched_address_gap_filler_enabled
+      WATCHED_ADDRESS_GAP_FILLER_INTERVAL: $watched_address_gap_filler_interval
```
Comment (nikugogoi) [inline docker/latest/docker-compose-ipld-eth-server.yml]: `WATCHED_ADDRESS_GAP_FILLER_ENABLED` and `WATCHED_ADDRESS_GAP_FILLER_INTERVAL` are not required in v3.2.2.
Watched addresses fill service was removed in previous [version](https://github.com/vulcanize/ipld-eth-server/releases/tag/v3.2.0)

**Review by nikugogoi — APPROVED** [review]

**Review by prathamesh0 — APPROVED** [review]

**Pull request merged**